lib/bitstyles_phoenix/component/icon.ex

```diff
@@ -99,5 +100,22 @@ defmodule BitstylesPhoenix.Component.Icon do
     opts
     |> Keyword.put_new(:width, @default_size)
     |> Keyword.put_new(:height, @default_size)
+    |> put_icon_file(Application.get_env(:bitstyles_phoenix, :icon_file, :inline))
```

Comment:
suggestion: Move the `put_new` here so you don't evaluate the app config unnecessarily

```elixir
|> put_new_lazy(:file, &icon_file_from_env/1)
```

Would probably mean to also write the `inline` option to `opts`

Reply:
I looked into this and I think it is not worth it. 
Passing in an `icon_file` here is mainly for making it testable. The default usecase will always be configuring it. Changing it in the way you suggested also makes us put `file: nil` for inline svgs (which might be an actual usecase).